decimal/bcd.go

```diff
@@ -256,5 +236,8 @@ func decodeStringFromBCD(bcdBuf []byte) (string, error) {
 		processNibble(lowNibble)
 	}
 
-	return bld.String(), nil
+	if bld.Len() == 0 || isNegative[sign] && bld.Len() == 1 {
```

Comment:
As far as I understand, we shouldn't validate anything for a negative scale since it is only for decode (and Tarantool doesn't expect to send invalid decimals), and in case of encode we just print trailing zeroes?

Reply:
Yes, it is.